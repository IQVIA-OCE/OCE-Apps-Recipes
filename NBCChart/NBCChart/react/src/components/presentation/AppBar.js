import React from "react";
import PropTypes from "prop-types";

import { Appbar as ApolloAppBar } from 'apollo-react-native';

const propTypes = {
    navigation: PropTypes.object,
    navigationOptions: PropTypes.object,
};

const AppBar = props => {
    const { navigation, navigationOptions } = props;

    return (
        <ApolloAppBar {...props}>
            { 
            navigation.isFirstRouteInParent() ? 
                null : 
                <ApolloAppBar.BackAction onPress={() => navigation.goBack()} />
            }

            <ApolloAppBar.Content
                title={navigationOptions.title}
            />
        </ApolloAppBar>
    )
}
    
AppBar.propTypes = propTypes;
  
export default AppBar;
