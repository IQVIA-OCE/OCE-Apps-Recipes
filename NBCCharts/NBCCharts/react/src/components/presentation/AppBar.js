import React from "react";
import PropTypes from "prop-types";
import { View } from "react-native";
import { Appbar as ApolloAppBar } from "apollo-react-native";

const propTypes = {
  navigation: PropTypes.object,
  navigationOptions: PropTypes.object
};

import { isIphone } from "../../constants";

const AppBar = props => {
  const { navigation, navigationOptions } = props;

  return (
    <ApolloAppBar {...props}>
      {navigation.isFirstRouteInParent() ? null : (
        <ApolloAppBar.Action onPress={() => navigation.goBack()} icon="arrow-left" />
      )}

      <ApolloAppBar.Content title={navigationOptions.title} />
    </ApolloAppBar>
  );
};

AppBar.propTypes = propTypes;

export default AppBar;
