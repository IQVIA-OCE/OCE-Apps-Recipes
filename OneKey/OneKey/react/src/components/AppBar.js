import React from "react";
import PropTypes from "prop-types";

import { Appbar as ApolloAppBar } from 'apollo-react-native';
import { useNavigationState, useRoute } from "@react-navigation/native";

const propTypes = {
    navigation: PropTypes.object,
    navigationOptions: PropTypes.object,
};

function useIsFirstRouteInParent() {
    const route = useRoute();
    const isFirstRouteInParent = useNavigationState(
      (state) => state.routes[0].key === route.key
    );

    return isFirstRouteInParent;
}

const AppBar = ({ navigation, scene, options, ...props }) => {
    const isFirstRouteInParent = useIsFirstRouteInParent();
    const title = options.headerTitle;

    const openValidationRequestForm = () => {
        const {navigate} = navigation;

        navigate('OneKeyValidationRequest');
    }

    return (
        <ApolloAppBar {...props}>
            {
                isFirstRouteInParent ?
                    null :
                    <ApolloAppBar.BackAction onPress={() => navigation.goBack()} />
            }

            <ApolloAppBar.Content
                title={title}
            />

            {
                title === "Accounts" ?
                    <ApolloAppBar.Action icon="account-plus" onPress={() => openValidationRequestForm()} /> : null
            }

            {/* {
                navigationOptions.title === "Create Validation Request" ?
                    <ApolloAppBar.Action icon="checkbox-marked-circle-outline"  onPress={navigationOptions.handleOneKeySubmit} /> : null
            } */}

        </ApolloAppBar>
    )
}

AppBar.propTypes = propTypes;

export default AppBar;
