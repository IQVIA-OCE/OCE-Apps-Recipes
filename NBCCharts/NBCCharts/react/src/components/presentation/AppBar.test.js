import { Appbar as ApolloAppBar } from "@oce-apps/apollo-react-native";
import React from "react";
import { render, fireEvent, screen } from "@testing-library/react-native";
import AppBar from "./AppBar";
import { useNavigationState, useRoute } from "@react-navigation/native";

jest.mock("@react-navigation/native", () => ({
  useRoute: jest.fn(),
  useNavigationState: jest.fn(),
}));

const createTestProps = (props) => ({
  ...props,
  options: {
    headerTitle: "Accounts",
  },
});

describe("AppBar", () => {
  beforeEach(() => {
    useRoute.mockReturnValue({ key: "Accounts" });
  });

  it("Should render AppBar component", () => {
    useNavigationState.mockReturnValueOnce(false);
    const props = createTestProps({ testID: "AppBarComponent" });
    render(<AppBar {...props} />);

    expect(screen.getByTestId("AppBarComponent")).toBeTruthy();
  });

  it("Should render AppBar component with isFirstRouteInParent", () => {
    useNavigationState.mockReturnValueOnce(true);
    const props = createTestProps({ testID: "AppBarComponent" });
    render(<AppBar {...props} />);

    expect(screen.getByTestId("AppBarComponent")).toBeTruthy();
  });

  it("Should render AppBar component with goBack action", () => {
    useNavigationState.mockReturnValueOnce(false);
    const goBack = jest.fn();
    const props = createTestProps({
      navigation: {
        goBack,
      },
    });

    render(<AppBar {...props} />);

    fireEvent.press(screen.UNSAFE_getAllByType(ApolloAppBar.Action)[0]);

    expect(goBack).toHaveBeenCalled();
  });
});
