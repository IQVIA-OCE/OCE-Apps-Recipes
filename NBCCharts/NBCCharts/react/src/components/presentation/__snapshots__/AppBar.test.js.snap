// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should render properly 1`] = `
<View
  navigation={
    Object {
      "goBack": [MockFunction],
      "isFirstRouteInParent": [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    }
  }
  navigationOptions={
    Object {
      "title": "Testing",
    }
  }
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0768FD",
      "elevation": 4,
      "flexDirection": "row",
      "height": 56,
      "paddingHorizontal": 4,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 4,
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessibilityTraits="button"
    accessible={true}
    centered={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "borderRadius": 0,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "paddingHorizontal": 12,
        },
        Array [
          Object {
            "marginLeft": 8,
          },
          Object {
            "alignItems": "center",
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      accessibilityRole="header"
      accessibilityTraits="header"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "fontSize": 17,
            },
            undefined,
          ],
        ]
      }
    >
      Testing
    </Text>
  </View>
  <View
    style={
      Object {
        "width": 48,
      }
    }
  />
</View>
`;

exports[`AppBar should render properly iPad version 1`] = `
<View
  navigation={
    Object {
      "goBack": [MockFunction],
      "isFirstRouteInParent": [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    }
  }
  navigationOptions={
    Object {
      "title": "Testing",
    }
  }
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0768FD",
      "elevation": 4,
      "flexDirection": "row",
      "height": 56,
      "paddingHorizontal": 4,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 4,
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessibilityTraits="button"
    accessible={true}
    centered={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "borderRadius": 0,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "paddingHorizontal": 12,
        },
        Array [
          Object {
            "marginLeft": 8,
          },
          Object {
            "alignItems": "center",
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      accessibilityRole="header"
      accessibilityTraits="header"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "fontSize": 17,
            },
            undefined,
          ],
        ]
      }
    >
      Testing
    </Text>
  </View>
  <View
    style={
      Object {
        "width": 48,
      }
    }
  />
</View>
`;
